From ae9dda7c2de2539c4a76e3164514592a90a034af Mon Sep 17 00:00:00 2001
From: Michal Hrusecky <michal.hrusecky@turris.com>
Date: Mon, 4 May 2026 13:10:08 +0200
Subject: [PATCH] python-package: Support optional removal in filepsec

Add option `--` that doesn't fail if file that irs requested for removal
doesn't exists.
---
 lang/python/python-package-install.sh | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/lang/python/python-package-install.sh b/lang/python/python-package-install.sh
index 4e7fb7af5..81524bbd9 100644
--- a/lang/python/python-package-install.sh
+++ b/lang/python/python-package-install.sh
@@ -97,6 +97,16 @@ while IFS='|' read -r cmd path file_mode dir_mode; do
 		rm -fR -- "$dest"/$path
 		;;
 
+	--)
+		log "Removing: \"$path\""
+
+		if ! path_exists "$dest" "$path"; then
+			log "\"$dest/$path\" not found"
+		fi
+
+		rm -fR -- "$dest"/$path
+		;;
+
 	=)
 		log "Setting recursive permissions \"${file_mode:-(none)}\"/\"${dir_mode:-(none)}\" on \"$path\""
 
-- 
2.54.0

